src/api/tasks.ts

```diff
@@ -1,37 +1,93 @@
 import { post, get, del, put } from "@/api";
+import { Duration } from "./duration";
 
-export interface Task {
+export type TaskID = number;
+
+export interface APITask {
   id: number;
-  duration: number;
   name: string;
-  description: string;
-  complete: boolean;
+  description: string | null;
+  duration: number | null;
+  completed: boolean;
+}
+
+export class Task {
+  public id: TaskID;
+  public name: string;
+  public duration?: Duration;
+  public description?: string;
+  public complete: boolean;
+
+  public constructor(
+    id: number,
+    name: string,
+    cfg: { duration?: Duration; description?: string; complete?: boolean }
```

Comment:
Could we possibly give a name to the type of `cfg`. It massively improves the error messages when it's given the wrong thing, which I am bound to do at some point

Reply:
But if I pass in an object with like `{'duration': ..., 'descirption': 'Complete task'}` or `{'description': 'Do a thing', 'complete': 0}` it'd be nice to get an error